packages/store/addon/-private/system/references/record.ts

```diff
@@ -17,12 +20,26 @@ type RecordInstance = import('../../ts-interfaces/record-instance').RecordInstan
    @extends Reference
 */
 export default class RecordReference extends Reference {
-  public get type() {
-    return INTERNAL_MODELS.get(this)!.modelName;
+  public get type(): string {
+    if (RECORD_ARRAY_MANAGER_IDENTIFIERS) {
+      let identifier = REFERENCE_CACHE.get(this) as StableRecordIdentifier;
+      return identifier.type;
+    } else {
+      return internalModelForReference(this)!.modelName;
+    }
   }
 
-  private get _id() {
-    return INTERNAL_MODELS.get(this)!.id;
+  private get _id(): string | null {
+    if (RECORD_ARRAY_MANAGER_IDENTIFIERS) {
+      let identifier = REFERENCE_CACHE.get(this) as StableRecordIdentifier;
+      if (identifier) {
+        return identifier.id;
+      }
+
+      return null;
+    } else {
+      return internalModelForReference(this)!.id;
+    }
```

Comment:
From meeting - an `identifier` getter probably makes sense.  Will discuss in next week's meeting.